### PR TITLE
Add package episode, finish modules

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
This explains modules and packages as well as using Revise.jl and PkgTemplates.jl
Closes https://github.com/carpentries-incubator/julia-novice/issues/8
Close https://github.com/carpentries-incubator/julia-novice/issues/9

### CODE REVIEW

File: _episodes/07-modules.md [hunk 1]
```diff
@@ -22,6 +22,72 @@ This is what it looks like:
...
+function shoot_distance(args...)
+    Trebuchets.shoot(args...)[2]
+end
+function shoot_distance(trebuchet::Trebuchet, env::Environment)
+    shoot_distance(env.wind, trebuchet.release_angle, trebuchet.counterweight)
+end
+
+function aim(trebuchet::Trebuchet, environment::Environment; ε = 1e-1, η = 0.05)
+    better_trebuchet = copy(trebuchet)
+    hit = x -> (shoot_distance([environment.wind, x[2], x[1]]) - environment.target_distance)
```
Comment (BeastyBlacksmith) [inline _episodes/07-modules.md:59]: @mose do you know an easy way to preserve `Trebuchet`s under arithmetic? That is if `Trebuchet` is an `AbstractArray` `Trebuchet(1,2) + [3,4]` returns a `Trebuchet`?

I don't really want to cover promotion rules, but probably thats the only way out?

Comment (giordano) [reply _episodes/07-modules.md:61]: What's `Trebuchets`?

File: _episodes/07-modules.md [hunk 2]
```diff
@@ -22,11 +24,77 @@ This is what it looks like:
 import Trebuchet as Trebuchets
 using ForwardDiff: gradient
 
+mutable struct Trebuchet <: AbstractVecor{Float64}
```
Comment (giordano) [inline _episodes/07-modules.md:27]:
```suggestion
mutable struct Trebuchet <: AbstractVector{Float64}
```

File: _episodes/07-modules.md (hunk 1 above)
Comment (giordano) [reply _episodes/07-modules.md:61]: Maybe you don't need promotion rules, but first of all you're likely going to have to implement the [array interface](https://docs.julialang.org/en/v1/manual/interfaces/#man-interface-array).  Then it's likely you can implement just `Base.:+(::Trebuchet, ::AbstractVector) = Trebuchet(...)`

Comment (BeastyBlacksmith) [reply _episodes/07-modules.md:61]: Thats three methods per operator for at least three operators. I guess that is also a bit too much.
Maybe the current version is not too bad after all

Comment (BeastyBlacksmith) [reply _episodes/07-modules.md:61]: I'll leave it for now

**Pull request merged**